### PR TITLE
Fixes #7619, APCs with no power will now not shock you

### LINKED ISSUE #7619: "APC zapped me, but dealt little to no damage as there was no wire leading to it, and no charge in the APC cell"
(WEB REPORT BY: dongilardoni REMOTE: 158.69.55.2:7777)
APC's that have no external power (not supplied electricity externally) without any battery inside it shouldn't zap the user no matter what hand protection they wear.

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: code/modules/power/apc.dm [hunk 1]
```diff
@@ -603,7 +603,7 @@
 			opened = 2
 			user.visible_message("<span class='warning'>The APC cover was knocked down with the [W.name] by [user.name]!</span>", \
 				"<span class='warning'>You knock down the APC cover with your [W.name]!</span>", \
-				"You hear bang")
+				"You hear a loud bang.") //"you hear bang" is so bad I have to leave a comment to immortalize it
```
Comment (PJB3005) [inline code/modules/power/apc.dm:606]: No code shitposting.

File: code/modules/power/apc.dm [hunk 2]
```diff
@@ -1337,4 +1337,12 @@ obj/machinery/power/apc/proc/autoset(var/val, var/on)
 		return 1
 	return 0
 
+//We kinda do things our own way and don't really use NOPOWER or such, so we need different sanity
+/obj/machinery/power/apc/shock(mob/user, prb, var/siemenspassed = -1)
+	if(shorted)
+		return
```
Comment (PJB3005) [inline code/modules/power/apc.dm]: This should return `FALSE` explicitly.

File: code/modules/power/apc.dm [hunk 3]
```diff
@@ -1337,4 +1337,12 @@ obj/machinery/power/apc/proc/autoset(var/val, var/on)
 		return 1
 	return 0
 
+//We kinda do things our own way and don't really use NOPOWER or such, so we need different sanity
+/obj/machinery/power/apc/shock(mob/user, prb, var/siemenspassed = -1)
+	if(shorted)
```
Comment (PJB3005) [inline code/modules/power/apc.dm]: Make this a single ìf.

**Review by PJB3005 — CHANGES_REQUESTED** [review]

File: code/modules/power/apc.dm (hunk 1 above)
Comment (Exxion) [reply code/modules/power/apc.dm:606]: This is a special case

Comment (PJB3005) [reply code/modules/power/apc.dm:606]: I disagree.

Comment (9600bauds) [reply code/modules/power/apc.dm:606]: Get off my back woman

File: code/modules/power/apc.dm (hunk 3 above)
Comment (9600bauds) [reply code/modules/power/apc.dm]: OK.

Why?

`if(shorted || (!cell && !charging)) ` is immediately less readable.

**Review by PJB3005 — APPROVED** [review]

File: code/modules/power/apc.dm (hunk 1 above)
Comment (PJB3005) [reply code/modules/power/apc.dm:606]: Fine.

File: code/modules/power/apc.dm (hunk 3 above)
Comment (PJB3005) [reply code/modules/power/apc.dm]: I disagree

**Pull request merged**